scripts/logs-backup-from-all-pods/extract_logs.py

```diff
@@ -0,0 +1,187 @@
+    # print("names: ", names)
+    return len(names)
+
+
+def get_no_of_active_users_count():
+    if days.strip() != 'None':
+        sql = "SELECT count(distinct  data->'auditData'-> 'source'->> 'name') FROM audit_events WHERE data->>'eventType' @@ ' AUTHENTICATION_SUCCESSFUL_AUDIT' AND created_at >= %s"
+        date = datetime.now() - timedelta(days=int(days))
+        cur_audit.execute(sql, [date])
+    else:
+        sql = "SELECT count(distinct  data->'auditData'-> 'source'->> 'name') FROM audit_events WHERE data->>'eventType' @@ ' AUTHENTICATION_SUCCESSFUL_AUDIT'"
+        cur_audit.execute(sql)
+    return cur_audit.fetchone()[0]
+
+
+def start_extraction():
+    try:
+        cookie = login_to_isd()
+        # get applications count
+        url = isd_gate_url + "/applications"
+        apps = get_response(url, cookie)
+        apps_count = len(apps)
+        # print("apps_count: ", apps_count)
+        pipelines_count = 0
+        for app in apps:
+            name = app['name']
+            url = isd_gate_url +  "/applications/" + name + "/pipelineConfigs"
+            pipelines = get_response(url, cookie)
+            pipelines_count = pipelines_count + len(pipelines)
+        # print("pipelines_count: ", pipelines_count)
+        cookie = login_to_isd()
+        url = isd_gate_url + "/credentials"
+        cloud_accounts = get_response(url, cookie)
+        cloud_accounts_count = len(cloud_accounts)
+        # print("cloud_accounts_count: ", cloud_accounts_count)
+        users = redis_conn.keys(
+            "spring:session:index:org.springframework.session.FindByIndexNameSessionRepository.PRINCIPAL_NAME_INDEX_NAME:*")
+        users_count = len(users)
+        # print("users_count: ", users_count)
+        no_of_users_executed_pipelines = get_pipelines_executed_users_count()
+        # print("no_of_users_executed_pipelines: ", no_of_users_executed_pipelines)
+        f = open("/tmp/logdir/usage_counts.txt", "w")
+        f.write("apps_count: "+ str(apps_count) + "\n")
+        f.write("pipelines_count: "+ str(pipelines_count)+ "\n")
+        f.write("cloud_accounts_count: "+ str(cloud_accounts_count)+ "\n")
+        f.write("users_count: "+ str(users_count)+ "\n")
+        if days.strip() != 'None':
+            dys = days
+        else:
+            dys = "all"
+        f.write("no_of_users_executed_pipelines: "+ str(no_of_users_executed_pipelines)+ " for " + dys + " days"+"\n")
+        if installation_type == 'ISD':
+            no_of_active_users_count = get_no_of_active_users_count()
+            f.write("no_of_active_users_count: "+ str(no_of_active_users_count)+ " for " + dys + " days"+"\n")
+            # print("no_of_active_users_count: ", no_of_active_users_count)
+        f.close()
+    except Exception as e:
+        print("Exception occurred while computing the application counts: ", e)
+        logging.error("Exception occurred while computing the application counts: ", exc_info=True)
+        raise e
```

Comment:
add a finally block and close all the connections in this block

Reply:
ok

Reply:
done